cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java

```diff
@@ -243,23 +247,44 @@ public void validateStorageTypeForGroup(AzureDiskType diskType, String flavor) {
     }
 
     @Retryable(backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000), maxAttempts = 5)
-    public List<CloudVmInstanceStatus> deallocateInstances(AuthenticatedContext ac, List<CloudInstance> vms) {
+    public List<CloudVmInstanceStatus> deallocateInstances(AuthenticatedContext ac, List<CloudInstance> vms) throws BatchInstanceActionFailedException {
```

Comment:
Please be so kind to split it into smaller methods for testability.

Reply:
Sure, I will do.